modules/core/src/smithy4s/Transformation.scala

```diff
@@ -119,4 +119,30 @@ object Transformation {
       }
     }
 
+  implicit def mappedHintsServiceTransform[Alg[_[_, _, _, _, _]]]: Transformation[Hints=>Hints, Service[Alg], Service[Alg]] =
```

Comment:
This could be improved, because `Hints => Hints` is not specific to Endpoints, and the instance is provided implicitly. Instead, a `class MapEndpointHints(f: Hints => Hints)` should be created, and the instance should be provided against that instead of the bare `Hints => Hints`, as that would let us add other types of endpoint mapping later on.

Reply:
Would something like this make sense, and  provide a transformation using EndpointMapper
```
sealed trait EndpointMapper
object EndpointMapper{
 case class MapEndpointHints(func: Hints => Hints) extends EndpointMapper

 }

Reply:
I don't think it does : it'll be dangerous to evolve by adding cases to that sealed trait (wrt exhaustive-pat-mats in client code), so it'd be preferable to not expose the members, and then if you're not exposing the members, why would you have a sealed at all ?